### PR TITLE
Feat/eslint@9

### PR DESCRIPTION
**Pull request by Charlie-Crowdstrike**
**This is a breaking change, and will require a major version bump. It will not be compatible with apps running < eslint `8.57.0`**

https://eslint.org/docs/latest/use/configure/migration-guide

* ran `npx @eslint/migrate-config .eslintrc.js`
* deleted `.eslintignore`
* edited generated `eslint.config.js` file to natively consume `index.js` config
* migrated `index.js` to new flat config format
* updated `package.json` dependencies to use flat config compatible packages, and added newly required packages for legacy plugin compatibility and configs

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
-    'plugin:n/recommended',
-  ],
-  plugins: [
-    'json-files',
-  ],
-  rules: {
-    'strict': 'error',
+const jsonFiles = require('eslint-plugin-json-files');
+const globals = require('globals');
+const { FlatCompat } = require('@eslint/eslintrc');
```
Comment (Charlie-Crowdstrike) [inline index.js:5]: eventually can get rid of `FlatCompat` once we update https://github.com/CrowdStrike/eslint-config-crowdstrike to flat configs as well

File: package.json [hunk 2]
```diff
@@ -32,19 +32,25 @@
...
-    "eslint-plugin-json-files": "^4.0.0",
+    "eslint-plugin-json-files": "^4.3.0",
     "eslint-plugin-n": "^17.0.0",
     "remark-cli": "^12.0.0",
     "remark-preset-lint-crowdstrike": "^4.0.0",
     "renovate-config-standard": "^2.0.0",
     "standard-version": "^9.0.0"
   },
   "peerDependencies": {
+    "eslint": ">=8.57.0",
```
Comment (Charlie-Crowdstrike) [inline package.json:53]: first version that supports flat configs

File: package.json [hunk 3]
```diff
@@ -32,21 +32,27 @@
...
     "eslint-config-crowdstrike": ">=1",
-    "eslint-plugin-json-files": ">=0.8",
+    "eslint-plugin-json-files": ">=4.3.0",
```
Comment (Charlie-Crowdstrike) [inline package.json:55]: pretty sure this is unncessary, as the same rule exists in `.8`

File: index.js [hunk 4]
```diff
@@ -1,21 +1,38 @@
...
-};
+const compat = new FlatCompat({
+  baseDirectory: __dirname, // optional; default: process.cwd()
+  resolvePluginsRelativeTo: __dirname, // optional
+  recommendedConfig: js.configs.recommended, // optional unless using "eslint:recommended"
+});
+
+module.exports = [
+  ...compat.extends('crowdstrike'),
+  nodePlugin.configs['flat/recommended-script'],
```
Comment (kellyselden) [inline index.js:17]: Looks like this was added in https://github.com/eslint-community/eslint-plugin-n/releases/tag/16.1.0, so we'll need to bump the peer dep.

File: package.json (hunk 3 above)
Comment (kellyselden) [reply package.json:55]: I'm fine with this.

File: package.json [hunk 5]
```diff
@@ -32,21 +32,27 @@
     "url": "https://github.com/CrowdStrike/eslint-config-crowdstrike-node/issues"
   },
   "homepage": "https://github.com/CrowdStrike/eslint-config-crowdstrike-node#readme",
+  "dependencies": {
+    "@eslint/eslintrc": "3.1.0",
```
Comment (kellyselden) [inline package.json:36]: We should probably put ^ on these.

File: index.js [hunk 6]
```diff
@@ -1,21 +1,38 @@
...
+  {
+    plugins: {
+      'json-files': jsonFiles,
+    },
+    processor: "json-files/json",
```
Comment (kellyselden) [inline index.js:22]: Not sure what's going on here. This should fail linting with https://github.com/CrowdStrike/eslint-config-crowdstrike/blob/main/index.js#L28?

Comment (Charlie-Crowdstrike) [reply index.js]: Fixed it by specifying those rules/processor only for `*.json` files

**Comment by kellyselden** [comment]
released eslint-config-crowdstrike-node@4.0.0-beta.0

**Pull request merged**